source/intel.js

```diff
@@ -1966,12 +1965,54 @@ NeptunesPride.npui.StarInspector = function () {
 
   return starInspector;
 };
+//Enable arrows keys to interact with editing a ship's instructions on stay.
+const fleet_order_keyboard_integeration = () => {
+  const event_name = "show_screen";
+
+  const selector_right_button =
+    "#contentArea > div > div.widget.fullscreen > div:nth-child(3) > div > div.widget.rel.col_black > div:nth-child(3)";
+  const selector_left_button =
+    "#contentArea > div > div.widget.fullscreen > div:nth-child(3) > div > div.widget.rel.col_black > div:nth-child(2)";
+  const selector_menu =
+    "#contentArea > div > div.widget.fullscreen > div:nth-child(3) > div > div.widget.rel.col_accent > div.widget.drop_down > select";
+
+  const apply_arrow_click = (element, selector, key) => {
+    console.log("Event added");
+    const NP_click = (event) => {
+      if (event.key === key) {
+        $(selector).mousedown().mouseup(); //NP does not respond to click
+        console.log(`${key} has been clicked!`);
+      }
+    };
+    element.keyup(NP_click);
+  };
+
+  const orig = { EditFleetOrder: NeptunesPride.npui.EditFleetOrder };
+  NeptunesPride.npui.EditFleetOrder = (config) => {
+    const efo = orig.EditFleetOrder(config);
+    efo.postRoost = () => {
+      // giving keyboard focus to the dropdown enables the following
+      // keyboard controls: up/down changes the selection, 'g' sets
+      // it to 'garrison', 'c' cycles through 'collect' options, and
+      // 'd' cycles through 'drop' options.
+      const menu = $(selector_menu);
+      menu.focus();
+    };
+    return efo;
+  };
+  apply_arrow_click(menu, selector_left_button, "ArrowLeft");
```

Comment:
This line is in the wrong spot. Not only does the variable `menu` not exist here, the element it is meant to refer to does not exist at this point in the flow, it won't be created until the screen is up, so `postRoost` is the right spot for this.

I don't remember how `Crux` destroys elements, but I believe it's an all new `menu` element every time, so just moving this line to after `menu.focus()` ought to be correct.

Reply:
I don't think `apply_arrow_click` needs to depend on `menu` and putting `apply_arrow_click` inside the `post_roost` callback makes it re-bind the hotkey every `post_roost` causing the event to be called many times. 

I found the left & right keys don't work when using the `menu.focus`. Although, the DropDown does crux object is not a normal dropdown menu so the text is a different field then the selector. If I can find a way to set the DropDown menu using any code I can make a solution & then clean up the optimal & minimal events.

Reply:
> I don't think `apply_arrow_click` needs to depend on `menu` and putting `apply_arrow_click` inside the `post_roost` callback makes it re-bind the hotkey every `post_roost` causing the event to be called many times.

OK we definitely saw some of the same behaviour but also some different behaviour. What version of Chrome?

For me, binding the keys inside post roost did, as you say, rebind the hotkeys every time. However, events were still called exactly once and everything worked as expected, including left/right arrows. 
 
> I found the left & right keys don't work when using the `menu.focus`. Although, the DropDown does crux object is not a normal dropdown menu so the text is a different field then the selector. If I can find a way to set the DropDown menu using any code I can make a solution & then clean up the optimal & minimal events.

I thought this was just because they weren't hooked up, I didn't realize this was happening even with the menu binding being redone in every roost.

Reply:
So there are two separate features (left&right) to navigate stars and (up&down) to navigate carrier function. Do either of these features work on your version of chrome?

I am running

Reply:
Sorry it took me a while to get back to this. For me, the branch from my original example does work completely, that is the left, right, up, down, and letter keys all function as expected. This is on MacOS with Chrome 108.0.5359.124 (Official Build) (arm64). The exact hash I was testing from is c131559865f8a5ac2217fab23a6b6552c9193f63.

I will fire up a virtual machine to test on Linux.

Reply:
On Chrome 108.0.5359.124 (64-bit) on Linux this same hash "almost" works. All the hotkeys cause the UI to react.

But pressing left or right to change stars _also_ changes the dropdown as though down arrow was pressed. This is pretty unexpected :-( 

The reason appears to be that in the basic linux keybindings for combo boxes, left and right function the same way as up and down. That is, if you turn off the extension and go to the dropdown with the tab key, the left and right arrow keys affect the dropdown selection. So both things are happening: it is changing stars, but also changing the drop down.

It turns out that if you handle `keydown` instead of `keyup` the behaviour is as expected. The existence of the event handler appears to disable the default handling of the key and everything works properly.